capstone/src/main/java/com/step902020/capstone/GcsStore.java

```diff
@@ -0,0 +1,74 @@
+package com.step902020.capstone;
+
+import com.google.cloud.storage.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.net.URL;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class GcsStore {
+
+  @Autowired
+  Storage storage;
+
+  /**
+   * generates a form for the user to submit profile picture
+   * @param projectId id of the current project
+   * @param bucketName bucket name preset to spring-bucket-jennysheng
+   * @param blobName name of the object matches user email
+   * @return html form in String
+   */
+  public String generateSignedPostPolicyV4(
```

Comment:
I'm a bit confused by this approach of generating the whole HTML form and using this PolicyV4 stuff. Can you just use a signed URL to which you can `PUT` an image, like described [here](https://github.com/spring-cloud/spring-cloud-gcp/issues/411#issuecomment-364949224)?
You should also be able to just change the `action` attribute of the form in JS rather than returning the whole HTML snippet for the form from the server.

Reply:
When I tried it without the PolicyV4 it gave an error saying that the identity of the user is anonymous. Also for the error that you got, can you tell me the size of the file that you tried to upload? I know this error exists and from what I can tell it seems like the error happens when the file is too small (smaller than 70kb, from what I can gather from testing various image sizes), but I could not figure out exactly what is causing the error. I will look into this more.

Reply:
Never mind. You can't really do a `PUT` from a form, and it won't be a simple change.